ocaml/xapi/xapi_network.ml

```diff
@@ -329,3 +329,62 @@ let with_networks_attached_for_vm ~__context ?host ~vm f =
         error "Caught %s while detaching networks" (string_of_exn e)
     end;
     raise e
+
+let assert_can_add_purpose ~__context ~network ~current porpoise =
+  let sop (*string-of-porpoise*) = Record_util.network_purpose_to_string in
```

Comment:
If we use `purpose` consistently everywhere that will make it easier to grep / read through the code 😄

Reply:
Yes, this is nothing to do with small whales either...

Reply:
Spoilsport.